### PR TITLE
feat(3075): Add pipeline level customization for pipeline template

### PR DESCRIPTION
**Pull request by klu909**
## Context

Add pipeline level customization for pipeline template
## Objective

Add pipeline level customization for pipeline template
## References

https://github.com/screwdriver-cd/screwdriver/issues/3077
https://github.com/screwdriver-cd/screwdriver/issues/3076
https://github.com/screwdriver-cd/screwdriver/issues/3080
https://github.com/screwdriver-cd/screwdriver/issues/3075
## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: lib/phase/merge.js [hunk 1]
```diff
@@ -40,16 +40,126 @@ async function fetchPipelineTemplate(
...
+        return !Object.entries(value).length;
+    }
+
+    return false;
+}
+
+/**
+ * Merge parsed yaml config with pipeline template config
  * Pipeline yaml will take precedence over pipeline template
  * @method mergeSetting
```
Comment (tkyi) [inline lib/phase/merge.js]: mergePipelineLevelConfig

File: lib/phase/merge.js [hunk 2]
```diff
@@ -40,16 +40,126 @@ async function fetchPipelineTemplate(
...
+        newPipeline.childPipelines = parsedDoc.childPipelines;
+    }
+
+    return newPipeline;
+}
+
+/**
+ * Merge parsed yaml config with pipeline template config
+ * Pipeline yaml will take precedence over pipeline template
+ * @method mergeSetting
```
Comment (tkyi) [inline lib/phase/merge.js]: mergeConfig

File: lib/phase/merge.js [hunk 3]
```diff
@@ -40,16 +40,126 @@ async function fetchPipelineTemplate(
...
+            scmUrl =>
+                !newScmUrls.find(
+                    newScmUrl => JSON.stringify(Object.keys(newScmUrl)) === JSON.stringify(Object.keys(scmUrl))
+                )
+        );
+
+        newPipeline.subscribe = {
+            ...templateSubscribeConfig,
+            ...yamlSubscribeConfig,
+            scmUrls: [...reducedOldScmUrl, ...newScmUrls]
```
Comment (VonnyJap) [inline lib/phase/merge.js:117]: Is there other subscribe config that we need to take care of? Looks like in this implementation, scmUrls is the only one being taken care of separately.

Comment (klu909) [reply lib/phase/merge.js:117]: for now it's only scmUrls, line 115 and 116 is just there to be kinda future proof. Can be removed if you want

Comment (VonnyJap) [reply lib/phase/merge.js:117]: it is ok, we can leave it there.

**Review by VonnyJap — APPROVED** [review]

**Pull request merged**